presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1011,6 +1045,23 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
                 return createAndAssignScope(table, scope, outputFields);
             }
 
+            Optional<ConnectorMaterializedViewDefinition> optionalMaterializedView = metadata.getMaterializedView(session, name);
```

Comment:
This is not the right place to inject the check. Will come back to this

Reply:
Why can't we do the check in `CreateMaterializedViewTask`?

Reply:
I will be doing it in StatementAnalyzer.visitCreateMaterializedView(), where it has more easy access to variables for semantic check.